src/acsets/DenseACSets.jl

```diff
@@ -470,22 +364,41 @@ ACSetInterface.has_subpart(acs::DynamicACSet, f::Symbol) =
   runtime(_has_subpart, acs.schema, f)
 
 @ct_enable function _has_subpart(@ct(S), @ct(f))
-  @ct s = SchemaDesc(S)
-  @ct f ∈ [s.homs; s.attrs]
+  @ct s = Schema(S)
+  @ct f ∈ arrows(s; just_names=true)
 end
 
-@inline ACSetInterface.incident(acs::SimpleACSet, part, f::Symbol) = preimage(acs.subparts[f], part)
-@inline ACSetInterface.incident(acs::SimpleACSet, parts::Union{AbstractVector,UnitRange}, f::Symbol) =
-  preimage_multi(acs.subparts[f], parts)
-@inline ACSetInterface.incident(acs::StructACSet{S}, ::Colon, f::Symbol) where {S} =
-  _incident(acs, Val{S}, :, Val{f})
+domain(acs::StructACSet{S}, f::Symbol) where {S} = _domain(acs, Val{S}, Val{f})
+domain(acs::DynamicACSet, f::Symbol) = runtime(_domain, acs, acs.schema, f)
 
-ACSetInterface.incident(acs::DynamicACSet, ::Colon, f::Symbol) where {S} =
-  runtime(_incident, acs, acs.schema, :, f)
+@ct_enable function _domain(acs, @ct(S), @ct(f))
+  @ct s = Schema(S)
+  parts(acs, @ct dom(s, f))
+end
+
+@inline ACSetInterface.incident(acs::SimpleACSet, part, f::Symbol; consistent=false) =
```

Comment:
Again, rather than calling this flag `consistent`, we should call it something meaningful and descriptive.